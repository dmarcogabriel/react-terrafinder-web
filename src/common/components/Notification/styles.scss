@import '~styles';

.Notification__container {
  position: fixed;
  top: 2rem;
  right: 2rem;
  left: 2rem;
  padding: 1rem;
  border-radius: 0.25rem;
  transition: opacity 5s;

  > p {
    opacity: 1;
    text-align: center;
  }
}

.success {
  background: $green-light;
  color: $green-dark;
}

.info {
  background: $blue;
  color: $blue-dark;
}

.warning {
  background: $primary-light;
  color: $orange;
}

.error {
  background: $red;
  color: $red-dark;
}
