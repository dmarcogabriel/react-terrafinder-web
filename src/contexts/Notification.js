import React, { createContext, useState, useEffect } from 'react';

export const Context = createContext();

export const NotificationProvider = ({ children }) => {
  const [show, setShow] = useState(false);
  const [notification, setNotification] = useState({
    message: '',
    type: 'info',
  });

  const showNotification = (message, type = 'info') => {
    setNotification({ message, type });
    setShow(true);
  };

  useEffect(() => {
    if (show) {
      setTimeout(() => {
        setShow(false);
      }, 5000);
    }
  }, [show]);

  return (
    <Context.Provider
      value={{
        show,
        notification,
        showNotification,
      }}
    >
      {children}
    </Context.Provider>
  );
};
