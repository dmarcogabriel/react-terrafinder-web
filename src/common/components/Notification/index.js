import React from 'react';
import { useNotification } from 'hooks/useNotification';
import './styles.scss';
import cn from 'classnames';

export default function Notification() {
  const { notification, show } = useNotification();

  const style = {
    opacity: show ? 0.8 : 0,
    zIndex: show ? -50 : 5000,
  };

  return (
    <div
      data-testid="notification"
      className={cn('Notification__container', notification.type)}
      style={style}
    >
      <p data-testid="notificationMessage">{notification.message}</p>
    </div>
  );
}
