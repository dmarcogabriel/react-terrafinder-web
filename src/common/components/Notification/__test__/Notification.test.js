import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
// import { render } from '@testing-library/react';
import Notification from '..';

describe('<Notification />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Notification />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  // Add tests here...

  it('matches snapshot', () => {
    const tree = renderer.create(<Notification />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
