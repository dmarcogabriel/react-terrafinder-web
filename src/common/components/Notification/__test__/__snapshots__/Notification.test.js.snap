// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Notification /> matches snapshot 1`] = `
<div
  className="Notification__container info"
  data-testid="notification"
  style={
    Object {
      "opacity": 0,
      "zIndex": 5000,
    }
  }
>
  <p
    data-testid="notificationMessage"
  >
    
  </p>
</div>
`;
